var/spack/repos/builtin/packages/pfunit/package.py

```diff
@@ -113,20 +113,26 @@ def cmake_args(self):
             self.define_from_variant("BUILD_SHARED", "shared"),
             "-DCMAKE_Fortran_MODULE_DIRECTORY=%s" % spec.prefix.include,
             self.define_from_variant("BUILD_DOCS", "docs"),
-            "-DMAX_ASSERT_RANK=%s" % spec.variants["max_array_rank"].value,
         ]
 
-        if self.spec.satisfies("%gcc@10:"):
-            args.append("-DCMAKE_Fortran_FLAGS_DEBUG=-g -O2 -fallow-argument-mismatch")
-
         if spec.satisfies("@4.0.0:"):
             args.append("-DSKIP_MPI=%s" % ("YES" if "~mpi" in spec else "NO"))
             args.append("-DSKIP_OPENMP=%s" % ("YES" if "~openmp" in spec else "NO"))
             args.append("-DSKIP_FHAMCREST=%s" % ("YES" if "~fhamcrest" in spec else "NO"))
             args.append("-DSKIP_ESMF=%s" % ("YES" if "~esmf" in spec else "NO"))
+            args.append("-DMAX_ASSERT_RANK=%s" % spec.variants["max_array_rank"].value)
         else:
+            if spec.satisfies("%gcc@10:"):
+                args.append("-DCMAKE_Fortran_FLAGS_DEBUG=-g -O2 -fallow-argument-mismatch")
```

Comment:
Why is this even setting -O2 in debug?!

Reply:
@germasch do you remember why you used O2 here?